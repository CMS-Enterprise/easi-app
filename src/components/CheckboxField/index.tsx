import React from 'react';
import classnames from 'classnames';

type CheckboxFieldProps = {
  checked?: boolean;
  disabled?: boolean;
  id: string;
  label: React.ReactNode;
  name: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onBlur: () => void;
  value: string;
  inputProps?: JSX.IntrinsicElements['input'];
  'data-testid'?: string;
};

const CheckboxField = ({
  checked,
  disabled,
  id,
  label,
  name,
  onChange,
  onBlur,
  value,
  inputProps,
  'data-testid': datatestid
}: CheckboxFieldProps) => {
  const checkboxClassNames = classnames('easi-checkbox', 'usa-checkbox', {
    'easy-checkbox--disabled': disabled
  });
  return (
    <div className={checkboxClassNames}>
      <input
        checked={checked}
        className="usa-checkbox__input"
        disabled={disabled}
        id={id}
        name={name}
        onChange={onChange}
        onBlur={onBlur}
        type="checkbox"
        value={value}
        {...inputProps}
        data-testid={datatestid}
      />
      <label className="usa-checkbox__label" htmlFor={id}>
        {label}
      </label>
    </div>
  );
};

export default CheckboxField;
