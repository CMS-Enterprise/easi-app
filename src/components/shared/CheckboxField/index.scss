.easi-checkbox {
  &--disabled {
    .easi-checkbox__input {
    }
  }
}
