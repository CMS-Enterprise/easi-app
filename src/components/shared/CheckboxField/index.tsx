import React from 'react';
import classnames from 'classnames';

type CheckboxFieldProps = {
  checked?: boolean;
  disabled?: boolean;
  id: string;
  label: string;
  name: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onBlur: () => void;
  value: string;
};

const CheckboxField = ({
  checked,
  disabled,
  id,
  label,
  name,
  onChange,
  onBlur,
  value
}: CheckboxFieldProps) => {
  const checkboxClassNames = classnames('easi-checkbox', 'usa-checkbox', {
    'easy-checkbox--disabled': disabled
  });
  return (
    <div className={checkboxClassNames}>
      <input
        checked={checked}
        className="usa-checkbox__input"
        disabled={disabled}
        id={id}
        name={name}
        onChange={onChange}
        onBlur={onBlur}
        type="checkbox"
        value={value}
      />
      <label className="usa-checkbox__label" htmlFor={id}>
        {label}
      </label>
    </div>
  );
};

export default CheckboxField;
