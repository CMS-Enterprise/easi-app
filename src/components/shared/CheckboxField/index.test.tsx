import React from 'react';
import { shallow } from 'enzyme';
import CheckboxField from './index';

describe('The Checkbox Field component', () => {
  it('renders without crashing', () => {
    shallow(
      <CheckboxField
        id="TestTextbox"
        label="Test Textbox"
        name="Test"
        onChange={() => {}}
        onBlur={() => {}}
        value="Test"
      />
    );
  });
});
