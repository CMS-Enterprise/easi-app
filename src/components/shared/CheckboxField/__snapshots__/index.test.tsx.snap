// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The Checkbox Field component > renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="easi-checkbox usa-checkbox"
  >
    <input
      class="usa-checkbox__input"
      id="TestTextbox"
      name="Test"
      type="checkbox"
      value="Test"
    />
    <label
      class="usa-checkbox__label"
      for="TestTextbox"
    >
      Test Textbox
    </label>
  </div>
</DocumentFragment>
`;
